packages/fxa-auth-server/lib/account-delete.ts

```diff
@@ -2,39 +2,55 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { CloudTasksClient } from '@google-cloud/tasks';
+import {
+  deleteAllPayPalBAs,
+  getAllPayPalBAByUid,
+} from 'fxa-shared/db/models/auth';
 import { StatsD } from 'hot-shots';
 import { Container } from 'typedi';
+
+import { CloudTasksClient } from '@google-cloud/tasks';
+import * as Sentry from '@sentry/node';
+
+import { ConfigType } from '../config';
 import DB from './db/index';
+import { ERRNO } from './error';
 import OAuthDb from './oauth/db';
 import { AppleIAP } from './payments/iap/apple-app-store/apple-iap';
 import { PlayBilling } from './payments/iap/google-play/play-billing';
 import { PayPalHelper } from './payments/paypal/helper';
 import { StripeHelper } from './payments/stripe';
+import { StripeFirestoreMultiError } from './payments/stripe-firestore';
+import pushBuilder from './push';
 import pushboxApi from './pushbox';
 import { accountDeleteCloudTaskPath } from './routes/cloud-tasks';
-import { AccountDeleteReasons, AppConfig, AuthLogger } from './types';
-
-import {
-  deleteAllPayPalBAs,
-  getAllPayPalBAByUid,
-} from 'fxa-shared/db/models/auth';
-import { ConfigType } from '../config';
-import * as Sentry from '@sentry/node';
-import { StripeFirestoreMultiError } from './payments/stripe-firestore';
+import { AppConfig, AuthLogger, AuthRequest } from './types';
 
 type FxaDbDeleteAccount = Pick<
   Awaited<ReturnType<ReturnType<typeof DB>['connect']>>,
-  'deleteAccount' | 'accountRecord' | 'account'
+  'deleteAccount' | 'accountRecord' | 'account' | 'devices'
 >;
 type OAuthDbDeleteAccount = Pick<typeof OAuthDb, 'removeTokensAndCodes'>;
 type PushboxDeleteAccount = Pick<
   ReturnType<typeof pushboxApi>,
   'deleteAccount'
 >;
+type PushForDeleteAccount = Pick<
+  ReturnType<typeof pushBuilder>,
+  'notifyAccountDestroyed'
+>;
+type Log = AuthLogger & { activityEvent: (data: Record<string, any>) => void };
 
-export type ReasonForDeletion = (typeof AccountDeleteReasons)[number];
-export type AccountDeleteOptions = { notify?: () => Promise<void> };
+export const enum ReasonForDeletion {
+  UserRequested = 'fxa_user_requested_account_delete',
+  Unverified = 'fxa_unverified_account_delete',
+  Cleanup = 'fxa_cleanup_account_delete',
+}
```

Comment:
What's the advantage of using enum?  Even TS docs are suggesting that we don't really need them: https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums ?

Reply:
Cool, I haven't seen the update here, nice to see it keeps changing! 😁 
Will rework to the guides recommendation.

Reply:
Enums are a contentious issue within TypeScript, which is why they've added this bit as an "option".

The biggest advantage of using enums is that it enforces that we don't have hanging magic values around our codebase, and requires that people use the enum rather than inserting those magic values or explicit strings.

Additionally, enums are easier & cleaner. From their own example, you also have to pull out the types every time you use the `as const` syntax.

> type Direction = typeof ODirection[keyof typeof ODirection];

This results in having both ODirection (the value) and Direction (the type).

I've seen far more arguments that make sense _for_ using enums than against. This isn't new, and has been a debate for years. TypeScript has also been adding new functionality to enums in the last few releases, which further cements enums as the right path forward imo. I'm a strong proponent of using enums over the `as const` syntax.